### PR TITLE
Model Dapr sidecar as an Aspire resource

### LINKED ISSUE #1553: Using WithEnvironment after dapr doesn't work
I need to reference an environment variable in Dapr's component configuration file, but after using the following code, I cannot get the variable in the dapr sidecar.

```csharp
var apiServiceSidecar = builder.AddProject<Projects.AspireApp_ApiService>("apiservice")
    .WithDaprSidecar(new DaprSidecarOptions
    {
        AppId = "apiservice",
        DaprGrpcPort = 50001,
        DaprHttpPort = 3500,
        ResourcesPaths = ["./components"],
    }).WithEnvironment("testtesttesttesttesttesttest", "testvalue");
```

For example, the secretstores.local.env component in dapr obtains values ​​from environment variables.

```
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: env-secret-store
spec:
  type: secretstores.local.env
  version: v1
```
```
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: mqtt-pubsub
spec:
  type: pubsub.mqtt3
  version: v1
  metadata:
    - name: url
      secretKeyRef:
        name: testtesttesttesttesttesttest
auth:
  secretStore: env-secret-store
```

Unable to extract value from testtesttesttesttesttesttest

### PR DESCRIPTION
**Pull request by philliphoff**
Currently, Dapr sidecars are modeled as annotations on a parent resource, which indicates the need for a Dapr sidecar started for that resource. The Dapr lifecycle hook will then generate appropriate Dapr CLI executable resources to start the sidecars.  When publishing, that hook will write Dapr sidecar resources to the manifest.

This model has a disadvantage in that, because the sidecars are not true resources, users cannot modify them like other resources (e.g. set environment variables). This was done originally due to the tight 1:1 mapping between Dapr sidecar and parent resource.  For example, it would be semantically invalid to create a Dapr sidecar resource and then reference it from several resources.

This PR updates the model such that the Dapr sidecar is a true resource, but one exposed only through the existing `AddDaprSidecar()` API to avoid semantic errors; users cannot create an arbitrary Dapr sidecar resource, only configure the one associated with a given resource.  The API allows the existing sidecar options (if any) to be specified, as well as use the existing resource annotation helpers, such as `WithEnvironment()`. 

The API looks like:

```csharp

builder.AddProject<Projects.DaprServiceA>("servicea")
       .WithDaprSidecar(
                 sidecarBuilder =>
                 {
                     sidecarBuilder
                         .WithOptions(new DaprSidecarOptions { ... })
                         .WithEnvironment("DAPR_TEST_VAR", "Dapr Test Value");
                 }
       )

```

> NOTE: Currently only the environment variables annotations are applied to the implicit Dapr CLI executable resource.

Resolves #1553
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/1604)

### CODE REVIEW

File: samples/dapr/AppHost/aspire-manifest.json [hunk 1]
```diff
@@ -32,6 +32,17 @@
         }
       }
     },
+    "servicea-dapr": {
+      "type": "dapr.v0",
+      "dapr": {
```
Comment (davidfowl) [inline samples/dapr/AppHost/aspire-manifest.json:37]: Why is this nested?

File: src/Aspire.Hosting.Dapr/DaprSidecarResource.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting.Dapr;
+
+/// <summary>
+/// Represents a Dapr sidecar resource.
+/// </summary>
+internal sealed class DaprSidecarResource : Resource, IDaprSidecarResource
```
Comment (davidfowl) [inline src/Aspire.Hosting.Dapr/DaprSidecarResource.cs:11]:
```suggestion
internal sealed class DaprSidecarResource(string name) : Resource(name), IDaprSidecarResource { }
```

Welcome to the future

**Comment by davidfowl** [comment]
We'll end up having 3 resources in the UI once we do this https://github.com/dotnet/aspire/issues/436

**Comment by philliphoff** [comment]
> We'll end up having 3 resources in the UI once we do this #436

I'm ok with that, though I'm curious as to what the UX for that will be like.  It'd be neat to be able to arrange the resources in some sort of hierarchical way.

**Review by davidfowl — APPROVED** [review]

**Comment by GFlisch** [comment]
Any idea when the preview3 with this feature will be available?

**Comment by davidfowl** [comment]
February is the next release. You can use ci builds to help verify this feature though.

**Comment by bit365** [comment]
thanks ^_^

**Pull request merged**